lib/replicache.dart

```diff
@@ -211,36 +343,13 @@ class Replicache implements ReadTransaction {
       _timer.cancel();
       _timer = null;
 
-      for (var i = 0;; i++) {
-        Map<String, dynamic> result = await _invoke(_name, 'pull', {
-          'remote': _remote,
-          'clientViewAuth': _clientViewAuth,
-        });
-        if (result.containsKey('error') &&
-            result['error'].containsKey('badAuth')) {
-          _reauthenticating = true;
-          print('Auth error: ${result['error']['badAuth']}');
-          if (getClientViewAuth == null) {
-            print('Auth error: getAuthToken is null');
-            break;
-          }
-          if (i == 2) {
-            break;
-          }
-          print('Refreshing auth token to try again...');
-          this._clientViewAuth = await getClientViewAuth();
-          _reauthenticating = false;
-        } else {
-          await _checkChange(result['root']);
-          break;
-        }
-      }
+      await _sync();
       _scheduleSync(5);
     } catch (e) {
       // We are seeing some consistency errors during sync -- we push commits,
```

Comment:
I don't understand this comment, probably is outdated. We do *not* necessarily expect that if we successfully push commit C and then pull a new client view that it includes C. We do not require the batch endpoint to operate synchronously or for it to be consistent with the clientview.

Reply:
Good point - @arv we can remove this entire catch block actually.